lib/puppet/type/user.rb

```diff
@@ -692,7 +692,7 @@ def delimiter
       defaultto false
     end
 
-    def eval_generate
+    def generate
       if !self[:purge_ssh_keys].empty? && self[:purge_ssh_keys] != :false
```

Comment:
I think `self[:purge_ssh_keys] != :false` can be removed because the parameter is now munged before we get to this code path, so it will always be `!= :false`